easybuild/framework/easyconfig/tools.py

```diff
@@ -206,7 +206,13 @@ def mk_node_name(spec):
     edge_attrs = [('style', 'dotted'), ('color', 'blue'), ('arrowhead', 'diamond')]
     for spec in specs:
         for dep in spec['ec'].all_dependencies:
-            dgr.add_edge((spec['module'], dep))
+            try:
+                dgr.add_edge((spec['module'], dep))
+            except Exception as e:
+                if "already in digraph" in str(e):
+                    continue
```

Comment:
@ocaisa At least log this properly, and add a comment above why we're allowing this?